theia-extensions/viewer-prototype/src/browser/trace-explorer/trace-explorer-sub-widgets/trace-explorer-views-widget.tsx

```diff
@@ -15,30 +17,31 @@ export class TraceExplorerViewsWidget extends ReactWidget {
     static LINE_HEIGHT = 16;
     static ROW_HEIGHT = (2 * TraceExplorerViewsWidget.LINE_HEIGHT) + TraceExplorerViewsWidget.LIST_MARGIN;
 
+    @inject(TspClientProvider) protected readonly tspClientProvider!: TspClientProvider;
     protected forceUpdateKey = false;
+    protected lastSelectedOutputIndex = -1;
 
-    protected outputAddedEmitter = new Emitter<OutputAddedSignalPayload>();
-    outputAddedSignal = this.outputAddedEmitter.event;
-
-    @inject(TraceExplorerOpenedTracesWidget) protected readonly openedTracesWidget!: TraceExplorerOpenedTracesWidget;
+    protected selectedExperiment: Experiment | undefined;
+    protected availableOutputDescriptors: OutputDescriptor[] | undefined;
 
     @postConstruct()
     init(): void {
         this.id = TraceExplorerViewsWidget.ID;
         this.title.label = TraceExplorerViewsWidget.LABEL;
-        this.toDispose.push(this.openedTracesWidget.availableOutputDescriptorsDidChange(() => {
-            this.update();
-        }));
-        this.toDispose.push(this.outputAddedEmitter);
+        signalManager().on(Signals.AVAILABLE_OUTPUTS_CHANGED, (payload: AvailableAnalysesChangedSignalPayload) => this.doHandleAvailableAnalysesChanged(payload));
         this.update();
     }
 
+    dispose(): void {
+        super.dispose();
+        signalManager().off(Signals.AVAILABLE_OUTPUTS_CHANGED, (payload: AvailableAnalysesChangedSignalPayload) => this.doHandleAvailableAnalysesChanged(payload));
```

Comment:
The listener does not get removed because it is not the same instance being passed that was added earlier.